[I 200713 09:35:29 inrepo:36] Testing notebook gradient_descent.ipynb
[W 200713 09:35:29 inrepo:48] No such kernel python2, falling back on kernel language=python
[W 200713 09:35:29 inrepo:60] Using kernel python3 to provide language: python
[I 200713 09:35:30 execute:404] Executing notebook with kernel: python3
Traceback (most recent call last):
  File "/src/inrepo.py", line 114, in <module>
    main()
  File "/src/inrepo.py", line 110, in main
    test_f(opts.test, opts.output_dir)
  File "/src/inrepo.py", line 75, in run_notebook
    nb, cwd=os.path.dirname(nb_path), kernel_name=kernel_name, timeout=600
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/nbconvert/preprocessors/execute.py", line 737, in executenb
    return ep.preprocess(nb, resources, km=km)[0]
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/nbconvert/preprocessors/execute.py", line 405, in preprocess
    nb, resources = super(ExecutePreprocessor, self).preprocess(nb, resources)
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/nbconvert/preprocessors/base.py", line 69, in preprocess
    nb.cells[index], resources = self.preprocess_cell(cell, resources, index)
  File "/srv/conda/envs/notebook/lib/python3.7/site-packages/nbconvert/preprocessors/execute.py", line 448, in preprocess_cell
    raise CellExecutionError.from_cell_and_msg(cell, out)
nbconvert.preprocessors.execute.CellExecutionError: An error occurred while executing the following cell:
------------------
def gradient_descent(f, start, max_steps=2000, epsilon=.000001):
    point = start
    grad = nd.Gradient(f)
    steps = 0
    result = 100
    d = -grad(point)
    point_list = [point.flatten().tolist()]
    while steps < max_steps and result > epsilon:
        steps += 1
        phi = lambda x: f((point + x * d).flatten())
        res = minimize_scalar(phi, method='brent')
        point += res.x * d
        point_list.append(point.flatten().tolist())
        d = -grad(point)
        result = np.linalg.norm(d)
    print "Step: {0}, Gradient Val: {1}, Point: {2}".format(steps, result, point)
    return point_list
------------------

[0;36m  File [0;32m"<ipython-input-3-89ed84177044>"[0;36m, line [0;32m16[0m
[0;31m    print "Step: {0}, Gradient Val: {1}, Point: {2}".format(steps, result, point)[0m
[0m                                                   ^[0m
[0;31mSyntaxError[0m[0;31m:[0m invalid syntax

SyntaxError: invalid syntax (<ipython-input-3-89ed84177044>, line 16)


Container exited with status: {'Error': None, 'StatusCode': 1}
